lib/routes-priv.js

```diff
@@ -91,12 +96,37 @@ class RoutesPriv {
       }
 
       const { user } = data
-      this._getTosEntry(user, res)
+      this._getTosEntry(user, (err, data) => {
+        if (err) {
+          log(err)
+          return sendGenericError(res)
+        }
+
+        return res.status(200).json(data)
+      })
+    })
+  }
+
+  _signupAffiliate (req, data, cb) {
+    const { code, user, codesrc } = data
+
+    const args = [{
+      user,
+      code,
+      codesrc
+    }]
+
+    const service = 'rest:core:affiliate'
+    const request = getGrenacheReq('signup', args, service)
+    request(req, {
+      json: (data) => {
```

Comment:
I might be missing something, could you clarify, please? I believe `json` is called for both error and success cases, shouldn't we check `data.success`?
https://github.com/bitfinexcom/bfx-lib-server-js/blob/master/src/express/grenacheClientService.js#L27